### ISSUE TITLE
False positive no-member on member of generic grandparent class

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#3131]
### Steps to reproduce

Run pylint on the following code:

```py
from typing import Generic, TypeVar

T = TypeVar('T')

class Base(Generic[T]):
    def foo(self) -> None:
        print('bar')

class Intermediate(Base[T]):
    pass

class Concrete(Intermediate[int]):
    pass

obj = Concrete()
obj.foo()
```

### Current behavior

The following message is reported:
```
E1101: Instance of 'Concrete' has no 'foo' member (no-member)
```

The message goes away if `Concrete` is made to inherit directly from `Base`, so without `Intermediate` in between them.

### Expected behavior

No message is reported.

### pylint --version output

pylint 2.4.0
astroid 2.3.0
Python 3.7.2 (default, Feb 26 2019, 13:02:33) 
[GCC 7.3.1 20180323 [gcc-7-branch revision 258812]]

### Notes

I wonder if this is related to #3129.

**Comment by PCManticore** [comment]
Thanks, I can reproduce it as well.

**Comment by jamesbraza** [comment]
I believe this is fixed as of pylint 2.7 for the provided repro above

**Comment by radkujawa** [comment]
I copied this example and tried pylint 2.7.2 and I still get this error.

**Comment by hippo91** [comment]
@jamesbraza @radkujawa both of you are right.
It was fixed  some times ago but we have been forced to revert the fixing commit (03d15b0f32f7d7c9b2cb062b9321e531bd954344) because it broke something else.
This will should be definitely fixed when PyCQA/astroid#927 will be merged.

**Comment by NeilGirdhar** [comment]
Should this be closed?  It appears to be fixed now.

**Comment by Pierre-Sassoulas** [comment]
Thank you for the reminder @NeilGirdhar, cleaning up issue is hard :)

**Comment by NeilGirdhar** [comment]
Least I can do to help.  Thank you for resolving these so quickly.
